### PR TITLE
Add new classes - CTPPSAlignmentCorrectionsData and PPSAlignmentConfi…

### PR DESCRIPTION
**Pull request by Kszemka**
#### PR description:
The main aim of this PR is to add new plots for CTPPS classes - CTPPSRPAlignmentCorrectionsData and PPSAlignmentConfiguration - to Payload Inspector software so to be able to visualize it on the website.

#### PR validation:
The code was tested by a script testPPSRPAlignmentCorrections.sh (which is mostly based on getPayloadData.py from CondCore/Utilities) just to check whether the code is properly added to PI software and generate some proper data in JSON files produced by PI.
In case of testing - testPPSRPAlignmentCorrections.sh script should be run with 'run' argument.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

No backport.

### CODE REVIEW

File: CondCore/CTPPSPlugins/interface/CTPPSRPAlignmentCorrectionsDataHelper.h [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            CTPPSRPAlignment::getStringFromRPEnum(rp) + " " + CTPPSRPAlignment::getStringFromShiftEnum(sh, unc) +
+                " [um] vs. Runs",
+            CTPPSRPAlignment::getStringFromRPEnum(rp) + " " + CTPPSRPAlignment::getStringFromShiftEnum(sh, unc) +
+                " [um]") {}
+
+  uint decodeRP(uint r) {
+    if (r == 3 || r == 23 || r == 103 || r == 123)
+      return r;
+    else {
+      CTPPSDetId* config = new CTPPSDetId(r);
```
Comment (tvami) [inline CondCore/CTPPSPlugins/interface/CTPPSRPAlignmentCorrectionsDataHelper.h:80]: is this `delete`-ed?

File: CondCore/CTPPSPlugins/plugins/PPSAlignmentConfiguration_PayloadInspector.cc [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ ****************************************************************************/
+
+#include "CondCore/Utilities/interface/PayloadInspectorModule.h"
+#include "CondCore/Utilities/interface/PayloadInspector.h"
+#include "CondCore/CondDB/interface/Time.h"
+#include "CondCore/CondDB/interface/PayloadReader.h"
+#include "CondFormats/PPSObjects/interface/PPSAlignmentConfiguration.h"
+#include "CondCore/CTPPSPlugins/interface/PPSAlignmentConfigurationHelper.h"
+
+namespace {
```
[the remark this answers is not in the document]
Comment (Kszemka) [reply CondCore/CTPPSPlugins/plugins/PPSAlignmentConfiguration_PayloadInspector.cc:14]: In my previous [PR](https://github.com/cms-sw/cmssw/pull/39024) PPSTimingCalibration_PayloadInspector was also formed with unnamed namespace - the same situation involves most of the PI classes, like [SiPixelPlugins](https://github.com/cms-sw/cmssw/blob/master/CondCore/SiPixelPlugins/plugins/SiPixel2DTemplateDBObject_PayloadInspector.cc#L49) or [BeamSpotPlugins](https://github.com/cms-sw/cmssw/blob/master/CondCore/BeamSpotPlugins/plugins/BeamSpotOnline_PayloadInspector.cc#L9) - this is why I wanted to keep these conventions. Should it be changed anyway?

[the remark this answers is not in the document]
Comment (mmusich) [reply CondCore/CTPPSPlugins/plugins/PPSAlignmentConfiguration_PayloadInspector.cc:14]: I cannot find any reference to having only named namespaces either [here](https://cms-sw.github.io/cms_coding_rules.html) or in the cpp core guidelines. Perhaps @tvami can clarify why that would be needed.

[the remark this answers is not in the document]
Comment (tvami) [reply CondCore/CTPPSPlugins/plugins/PPSAlignmentConfiguration_PayloadInspector.cc:14]: > or in the cpp core guidelines

http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rs-unnamed

Matti reported a problem with this recently here:
https://github.com/cms-sw/cmssw/issues/38977

[the remark this answers is not in the document]
Comment (mmusich) [reply CondCore/CTPPSPlugins/plugins/PPSAlignmentConfiguration_PayloadInspector.cc:14]: excepted this is not a header...

[the remark this answers is not in the document]
Comment (tvami) [reply CondCore/CTPPSPlugins/plugins/PPSAlignmentConfiguration_PayloadInspector.cc:14]: ahh yes, that's right

**Comment by Kszemka** [comment]
The above mentioned issues were fixed :)

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
+db
 * tests pass

**Comment by mmusich** [comment]
>  tests pass

though the test is actually [not testing anything](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c0057b/27291/unitTests/src/CondCore/CTPPSPlugins/test/testPPSTimingCalibration/testing.log)

https://github.com/cms-sw/cmssw/blob/a9b836229a8b2869e3d342b70dd1332e45e0681a/CondCore/CTPPSPlugins/test/testPPSTimingCalibration.cpp#L28-L44

why is the test configured like that?

**Comment by perrotta** [comment]
@Kszemka please take a look at https://github.com/cms-sw/cmssw/pull/39274#issuecomment-1235601524

**Comment by Kszemka** [comment]
The test is working similar to testPPSTimingCalibration.sh but it gives an option to choose tag from the database and runs without changing the code (given as argv[] like this - 
image)... although it was not added in this PR, already merged in one of the previous [ones](https://github.com/cms-sw/cmssw/pull/38915/files#diff-6660e1e20a272eefa2523e65f65348cc7126f626e788883494dc8ef98dfe71d2)

**Comment by mmusich** [comment]
can `testPPSTimingCalibration.sh` become part of the unit test suite?

**Comment by Kszemka** [comment]
testPPSTimingCalibration.sh also demand some arguments to run it with - maybe the better option is just to change the testPPSTimingCalibration.cpp and also provide inside the code example tag and runs so it will work with _scram b runtests_?

**Comment by mmusich** [comment]
> and also provide inside the code example tag and runs so it will work with scram b runtests?

Yes that was my initial comment

**Comment by Kszemka** [comment]
Changed testPPSTimingCalibration.cpp so it works with scram b runtests, also added similar testPPSRPAlignmentCorrections.cpp for the new class that was added with this PR.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by tvami** [comment]
+db
 * unit tests pass https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c0057b/27340/unitTests/success.html

**Comment by perrotta** [comment]
@Kszemka I'm just curious: why did you name the CTPPSRPAlignmentCorrectionsDataHelper with initial "CTPPS", while the other class introduced here, and the other one already present in CondCore/CTPPSPlugins start with "PPS" (which is, as far as I remember, the agreed naming convention for this code nowadays)?

**Comment by Kszemka** [comment]
It is simply connected with the names of the classes that are already used in db and also are used for the PI visualization - there is [CTPPSRPAlignmentCorrectionsData](https://github.com/cms-sw/cmssw/blob/master/CondFormats/PPSObjects/src/CTPPSRPAlignmentCorrectionsData.cc) and [PPSAlignmentConfiguration](https://github.com/cms-sw/cmssw/blob/master/CondFormats/PPSObjects/src/PPSAlignmentConfiguration.cc) - thought it would be easier if the names somehow corresponds to each other. Should I change the name of CTPPSRPAlignmentCorrectionsData to PPS at the beginning anyway?

**Comment by perrotta** [comment]
> @perrotta, it is simply connected with the names of the classes that are already used in db and also are used for the PI visualization - there is [CTPPSRPAlignmentCorrectionsData](https://github.com/cms-sw/cmssw/blob/master/CondFormats/PPSObjects/src/CTPPSRPAlignmentCorrectionsData.cc) and [PPSAlignmentConfiguration](https://github.com/cms-sw/cmssw/blob/master/CondFormats/PPSObjects/src/PPSAlignmentConfiguration.cc) - thought it would be easier if the names somehow correspond to each other. Should I change the name of CTPPSRPAlignmentCorrectionsData to PPS at the beginning anyway?

Ok, it can make sense.
I think that at some point it was decided to migrate all "CTPPS" in the class names into "PPS", and I also think that it should be done at some point. But for now, let stick on the same names of the classes that are already used in db. Until we'll eventually migrate everything it will be easier to match each other as such.

**Comment by perrotta** [comment]
+1

**Pull request merged**